File: UICollectionView_p1_ObjC/UICollectionView_p1_ObjC/Models/GalleryItem.m
//
//  GalleryItem.m
//  UICollectionView_p1_ObjC
//
//  Created by olxios on 19/11/14.
//  Copyright (c) 2014 olxios. All rights reserved.
//

#import "GalleryItem.h"

@implementation GalleryItem

+ (instancetype)galleryItemWithDictionary:(NSDictionary *)dictionary
{
    GalleryItem *item = [[GalleryItem alloc] init];
    
    item.itemImage = dictionary[@"itemImage"];
    
    return item;
}

@end

File: UICollectionView_p1_Swift/UICollectionView_p1_Swift/Models/GalleryItem.swift
//
//  GalleryItem.swift
//  UICollectionView_p1_Swift
//
//  Created by olxios on 20/11/14.
//  Copyright (c) 2014 olxios. All rights reserved.
//

import Foundation

class GalleryItem {
    
    var itemImage: String
    
    init(dataDictionary:Dictionary<String,String>) {
        itemImage = dataDictionary["itemImage"]!
    }
    
    class func newGalleryItem(dataDictionary:Dictionary<String,String>) -> GalleryItem {
        return GalleryItem(dataDictionary: dataDictionary)
    }
    
}
